UKHomeOffice/nmsw-ui
# NMSW-276 Upload passenger file (FAL 6)

# Ticket

NMSW-276

----

## AC

Scenario 1: Selected ‘Yes’ and clicked Save and Continue

Given I have selected yes on the Will you be uploading passenger details
When I have clicked Save and Continue
Then I am shown the Upload the passenger details (FAL 6) page

Scenario 2: Click on Check for errors button

Given I have uploaded a file
And There are no errors
When I click check for errors
Then I am shown the no errors found page

ACTIONS:

Given a user is on the 'Will you be uploading passenger details' page
When a user clicks 'Yes'
And they click Save and Continue
Then they are shown the Upload the passenger details (FAL 6) page

Given a user is on the 'Will you be uploading passenger details' page
When a user clicks 'No'
And they click Save and Continue
Then they are shown the task list page

Given a user has uploaded a file
When a user clicks check for errors
Then the FE sends a POST to the /declaration/{declarationId}/upload-fal6 endpoint

Given the FE has sent a POST to the /declaration/{declarationId}/upload-fal6 endpoint
When the response is a 200
Then the FE shows the no errors found page

ERRORS:

Given a user has not uploaded a file
When they click check for errors
Then a form field styled error is shown

Given a user has uploaded a file not of type .csv or .xlsx
When they click check for errors
Then a form field styled error is shown

Given a user has uploaded a file with a size greater than 4MB
When they click check for errors
Then a form field styled error is shown

----

## To test

- Sign in
- Go to draft report if one exists
- Click start a new report
- Upload a General Declaration FAL 1
- from the task list page click on Upload Passenger Details
- > You should be taken to the 'Will you be uploading passenger details' page
- Select Yes
- Click Save and continue
- You should be taken to the Upload Passenger Details (FAL 6) page
- Click the check for error button
- You should see an error of “Select a Passenger details (FAL 6)”
- Upload an image file
- Click on `Check for errors`
- > Error of “The selected file must be a csv or xlsx” should be shown
- Upload a large file (over 4MB)
- Click on `Check for errors`
- > Error of “The file must be smaller than 4MB” should be shown
- Click on `The file must be smaller than 4MB` link in the `Error summary panel`
- You should be scrolled to the Upload a file question
- Upload a new file
- The error message should clear

For dev testing 404
- In `FileUploadForm` change the endpoint to `/declaration//upload-fal6`
- Upload a file
- Click `Check for errors`
- > You should be taken to something went wrong page

SUCCESS SCENARIO:
- Upload the a good Passenger Details FAL 6 file
- Click `Check for errors`
- You should be taken to the confirmation of upload page with a success message
- Click `Save and continue`
- > You should be taken to the task details page

----

## Notes

- There will be no updates to the task list here as currently we don't have the get endpoint for FAL6 - this will be updated later on
- Currently the BE will accept any file (even a blank FAL 6 form) as cell validation has not been implemented yet
